onnxruntime/core/providers/cuda/tensor/slice.cc

```diff
@@ -63,46 +63,43 @@ REGISTER_V11_TYPED_SLICE(int32_t)
 REGISTER_V11_TYPED_SLICE(int64_t)
 REGISTER_V11_TYPED_SLICE(float)
 
-template <bool dynamic>
-Status Slice<dynamic>::ComputeInternal(OpKernelContext* ctx) const {
-  const Tensor* input_tensor = GetSlicedOrUnslicedTensor(ctx);
+static Status SliceImpCore(const void* input_data, void* output_data,
+                           size_t element_size, size_t dimension_count,
+                           const TArray<int64_t>& starts_buffer, const TArray<int64_t>& steps_buffer,
+                           const TArray<int64_t>& input_strides, const TArray<fast_divmod>& output_strides,
+                           TensorShape& output_shape) {
+  if (output_shape.Size() == 0) {
+    return Status::OK();
+  }
 
-  ORT_ENFORCE(nullptr != input_tensor);
+  return SliceImpl(element_size,
+                   gsl::narrow_cast<int32_t>(dimension_count),
+                   starts_buffer,
+                   steps_buffer,
+                   input_strides,
+                   output_strides,
+                   input_data,
+                   output_data,
+                   output_shape.Size());
+}
 
-  auto& input_dimensions = input_tensor->Shape().GetDims();
+namespace SliceCuda {
 
-  // Initialize the starts & ends to the actual tensor shape
+static Status ComputeSliceStrides(const TensorShape& input_shape,
+                                  TArray<int64_t>& input_strides,
```

Comment:
To avoid code duplication - shared code between the stand-alone Slice impl (`SliceCuda::Impl()`) and the CUDA Slice kernel (ComputeInternal())